README.md

```diff
@@ -117,6 +117,26 @@ containers:
       value: "SIGTERM"
 ```
 
+### Cooldown
+
+By default `OOMHero` sends one signal per second to other processes once they reach warning or critical threshold.
+This might be undesireable behavior in some circumstances, therefore cooldown can be consfigured.
+Once set, signal will be sent no more often than once in `cooldown` for each signal type separately.
+In other words other processes would not receive more than one warning and one ciritcal signal more often than once in `cooldown`.
+
+To configure cooldown set `COOLDOWN` environment variable in deployment definition to a desired number of seconds:
```

Comment:
I think this is a great idea!

However, I'd like to test whether this parsing actually works (and for anti-regression). Therefore it would be best not to read environment variables in `init` function, but move the reading to the beginning of `watchProcesses` as this would allow re-setting this variable in tests. To allow testing `CRITICAL` and `WARNING` variables, I'd move the reading them there as well, effectively getting rid of `init` function. What do you think?

Reply:
I liked it. Can't we have this logic on `main()` instead ?

Reply:
There is a way to do it, but it makes the tests more brittle, more on it later.

`watchProcesses` function accepts arguments that are abstract:
- they can be stubs in tests
- and actual implementations in `main`
 
Therefore tests can't test anything that is directly in `main` as this is how the real application starts, I can't stub anything in there. Each test runs `watchProcesses` as "system under test" with a stubbed timer and processes getter.

What could be done (but make tests more brittle) is to have a function that reads environment variables and call it from both `main` (directly) and in each test that needs to reload environment variables. It is feasible, but it has a drawback in that tests should resemble the real thing as much as possible and this introduces a small difference. The bad effect it could have is: somebody might remove the reading from `main` and tests would still work. If variables were read in `watchProcesses`, removing the read from there would make tests fail.

Reply:
On that case let's just use your approach then, implement it on `watchProcesses`.